server/pbench/bin/pbench-move-unpacked

```diff
@@ -0,0 +1,181 @@
+
+    # make sure that all the relevant state directories exist
+    mk_dirs $hostname
+    # ... and a couple of other necessities.
+    if [[ $? -ne 0 ]] ;then
+        echo "$TS: Creation of $hostname processing directories failed: code $status" >&4
+        nerrs=$nerrs+1
+        continue
+    fi
+    mkdir -p $INCOMING/$hostname
+    if [[ $? -ne 0 ]] ;then
+        echo "$TS: Creation of $INCOMING/$hostname failed: code $status" >&4
+        nerrs=$nerrs+1
+        continue
+    fi
+    
+    # XXXX - for now, if it's a duplicate name, just punt and avoid producing the error - the full
+    # solution will involve renaming the unpacked directory appropriately.
+    if [ ${resultname%%.*} == "DUPLICATE__NAME" ] ;then
+        ndups=$ndups+1
+        continue
+    fi
+
+    incoming=$INCOMING/$hostname/$resultname
+    mkdir -p $incoming
+    status=$?
+    if [[ $status -ne 0 ]] ;then
+        echo "$TS: mkdir -p $incoming failed: code $status" >&4
+        nerrs=$nerrs+1
+        continue
+    fi
+
+    let start_time=$(date +%s)
+    PROG="pbench-unpack-tarballs"
+    pushd $TMP_PATH/$PROG/$hostname > /dev/null 2>&1
+    popd > /dev/null 2>&1
+
+    # copy the tarball contents to INCOMING - remove the link on failure
+    cp -R $TMP_PATH/$PROG/$hostname/$resultname $incoming.copy
+    status=$?
+    if [[ $status -ne 0 ]] ;then
+        echo "$TS: Cannot copy $TMP_PATH/$PROG/$hostname/$resultname to $incoming.copy: code $status" >&4
+        rm $RESULTS/$hostname/$prefix$resultname
+        nerrs=$nerrs+1
+        continue
+    fi
+
+    # remove the symlink from incoming
+    rm $incoming
+    status=$?
+    if [[ $status -ne 0 ]] ;then
+        echo "$TS: Cannot remove $incoming: code $status" >&4
+	rm $RESULTS/$hostname/$prefix$resultname
+        nerrs=$nerrs+1
+        continue
+    fi
+
+    # rename the copied file to its original name
+    mv $incoming.copy $incoming
+    status=$?
+    if [[ $status -ne 0 ]] ;then
+        echo "$TS: Cannot rename $incoming.copy to $incoming: code $status" >&4
+	rm $RESULTS/$hostname/$prefix$resultname
+        nerrs=$nerrs+1
+        continue
+    fi
+
+    # remove the unpacked tarballs from TMP_PATH directory
+    rm -R $TMP_PATH/$PROG/$hostname/$resultname
+    status=$?
+    if [[ $status -ne 0 ]] ;then
+        echo "$TS: Cannot remove $TMP_PATH/$PROG/$hostname/$resultname: code $status" >&4
+        nerrs=$nerrs+1
+        continue
+    fi
+
+    # move the link to $linkdest directory
+    mv $result $(echo $result | sed "s/$linksrc/$linkdest/")
+    status=$?
+    if [[ $status -ne 0 ]] ;then
+        echo "$TS: Cannot move $result link from $linksrc to $linkdest: code $status" >&4
+        rm $RESULTS/$hostname/$prefix$resultname
+        nerrs=$nerrs+1
+        continue
+    fi
+    let end_time=$(date +%s)
+    let duration=end_time-start_time
+    # log the success
+    echo "$TS: $hostname/$resultname: success - elapsed time (secs): $duration - size (bytes): $size"
+}
```

Comment:
If there is manual cleanup to be done, there should be some instructions appended to a file that can then be included in the mail  at the end of the script.